crates/polars-core/src/datatypes/static_array.rs

```diff
@@ -76,6 +77,13 @@ pub trait StaticArray:
     fn from_zeroable_vec(v: Vec<Self::ZeroableValueT<'_>>, dtype: DataType) -> Self {
         v.into_iter().collect_arr_with_dtype(dtype)
     }
+
+    fn full_null(length: usize, dtype: DataType) -> Self {
+        unsafe {
+            std::ptr::read(Box::into_raw(new_null_array(dtype.to_arrow(), length))
+                as *const dyn Array as *const Self)
```

Comment:
This is not what I had in mind 😅 . When I said move it to `StaticArray` I meant an implementation for each array type without any casts/unsafe code.

Reply:
I think every implementation is just
```rust
fn full_null(length: usize, dtype: DataType) -> Self {
    Self::new_null(dtype.to_arrow(), length)
}
```

Just copy/paste that into each implementation in `static_array.rs` with `fn full_null(len: usize, dtype: DataType) -> Self;` in the trait.